### PR TITLE
Add support for bundling using Rollup

### PR DESCRIPTION
**Pull request by andrewiggins**
Couple of decisions I made up I wanted to call out:

1. I decided to include `node-resolve` and `commonjs` plugins as deps so our default rollup config is mildly useful. Was thinking for uses with `microbundle` where there isn't a `rollup.config.js` for us to use. Would be curious to get other's thoughts here on what plugins we should include.

2. If a rollup config is present, then I just use that unmodified. I'm not sure if rollup provides an API to modify the config/options of plugin instances from a config object. I think this method (of just using the provided config unmodified) is fine though. I think the only thing lost right now is code coverage (though I guess I could just add another babel plugin instance with only the coverage plugin configured...)

This is still a work in progress but thought I'd put an early draft out here to get some feedback. I also need to update the testing infra a bit so I can validate this implementation but that's gonna come in a different PR.

### CODE REVIEW

File: src/lib/babel.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
+/**
+ * @param {import('../configure').Options} options
+ */
+export function babelConfig(options) {
```
Comment (andrewiggins) [inline src/lib/babel.js:4]: This file is basically the same as babel-loader.js except I separated the config part from the loader part so I could re-use it in rollup.

File: src/rollup.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+	} else if (rollupConfig && rollupConfig.then) {
+		rollupConfig = null;
+		console.warn(
+			chalk.yellow(
+				`Karmatic does not currently support asynchronous rollup configs. Using a default config instead.`
+			)
+		);
+	}
+
+	return rollupConfig || getDefaultConfig(options);
```
Comment (andrewiggins) [inline src/rollup.js:75]: Note to self: Consider adding code coverage babel plugin here if using existing config

**Comment by marvinhagemeister** [comment]
As usual, this is fantastic work 🙌 

1. Agree, I'm in favor of adding both `@rollup/plugin-node-resolve` and `@rollup/plugin-commonjs`. Those are very much needed in nearly every project anyway.

2. Never seen such an API, too. The docs don't seem to mention anything like that.

**Comment by developit** [comment]
Nice!

Rollup does provide [an API](https://rollupjs.org/guide/en/#options) to modify the config, but plugin "options" are already consumed at that point (since they're just parameters to functions not called by rollup):

```js
rollup.rollup({
    plugins: [
        {
            name: 'modify-config-plugin',
            options(inputOptions) {
                inputOptions.plugins[0]  // instance of the plugin
            }
        }
    ]
});
```

File: src/rollup.js (hunk 2 above)
Comment (andrewiggins) [reply src/rollup.js]: Done

**Comment by andrewiggins** [comment]
Okay, I think this is ready to go! The rollup testing is a little weird since we include rollup in karmatic as a devDep through microbundle, so technically the tests run using that version of rollup and not the one specified in the test package.json but oh well 🤷‍♂️ - it's good enough I think. Was still useful for me to catch some bugs in properly loading rollup configs.

File: README.md [hunk 3]
```diff
@@ -1,12 +1,12 @@
 # Karmatic [![npm](https://img.shields.io/npm/v/karmatic.svg)](https://npm.im/karmatic) [![travis](https://travis-ci.org/developit/karmatic.svg?branch=master)](https://travis-ci.org/developit/karmatic)
 
-A simplified zero-configuration wrapper around [Karma], [Webpack], [Jasmine] & [Puppeteer].
+A simplified zero-configuration wrapper around [Karma], [Rollup] or [Webpack], [Jasmine] & [Puppeteer].
```
Comment (andrewiggins) [inline README.md:3]: Any preferences on how to talk about Rollup here?

Comment (developit) [reply README.md:3]:
```suggestion
Zero-config browser tests powered by [Karma] & [Puppeteer], with automatic [Rollup]/[Webpack] support.
```

**Pull request merged**